### PR TITLE
cmd/uses: extract function `intersection_of_dependents`

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

~~This also converts `use_runtime_dependents` into an instance variable, similar to `cmd/deps.rb`'s `@use_runtime_dependencies`~~

Confirmed working as before with `libzip`:

<details>
<summary><code>brew uses libzip</code></summary>
<br/>

```console
$ brew uses libzip | cat                            
ebook-tools
freedink
fuse-zip
huexpress
ideviceinstaller
katago
mgba
openrct2
osrm-backend
php
php@7.2
php@7.3
planck
ppsspp
tldr
ykneomgr
```

</details>

<details>
<summary><code>brew uses libzip --recursive</code></summary>
<br/>

```console
$ brew uses libzip --recursive | cat
deployer
easyengine
ebook-tools
freedink
fuse-zip
huexpress
ideviceinstaller
katago
mgba
openrct2
osrm-backend
php
php@7.2
php@7.3
phplint
planck
ppsspp
tldr
ykneomgr
```

</details>

(This was originally part 2 of #8152 but is now extracted out as its own pull request)

### CODE REVIEW

File: Library/Homebrew/cmd/uses.rb [hunk 1]
```diff
@@ -62,19 +62,28 @@ def uses
       args.named.map { |name| OpenStruct.new name: name, full_name: name }
     end
 
-    use_runtime_dependents = args.installed? &&
-                             !args.include_build? &&
-                             !args.include_test? &&
-                             !args.include_optional? &&
-                             !args.skip_recommended?
+    @use_runtime_dependents = args.installed? &&
```
Comment (reitermarkus) [inline Library/Homebrew/cmd/uses.rb:65]: Pass this as an argument instead of making it an instance variable.

**Comment by reitermarkus** [comment]
Do not merge until https://github.com/Homebrew/brew/pull/8144 is merged.

**Comment by reitermarkus** [comment]
This needs a rebase now.

**Comment by SeekingMeaning** [comment]
Hm, I rebased but there aren't any conflicts

**Comment by SeekingMeaning** [comment]
https://github.com/Homebrew/brew/pull/8158/checks?check_run_id=933221748#step:12:70

```
     Failure/Error:
       expect { brew "uses", "--recursive", "foo" }
         .to output(/(bar\nbaz|baz\nbar)/).to_stdout
         .and not_to_output.to_stderr
         .and be_a_success

       expected block to output /(bar\nbaz|baz\nbar)/ to stdout, but output "bar\n"
```

Need to add `args` parameter
